contracts/src/Assets.sol

```diff
@@ -73,15 +84,19 @@ library Assets {
             revert InvalidDestination();
         }
 
-        _transferToAgent(assetHubAgent, token, sender, amount);
+        if (tokenOwnerAgentID == bytes32(0)) {
+            _transferToAgent(agent, token, sender, amount);
+        } else {
+            _burn(agentExecutor, agent, token, amount);
+        }
```

Comment:
Would prefer add another external function to distinguish the two cases? (e.g. `sendNativeToken` vs `repayForeignToken` more explicitly).

Reply:
I think this would make the user experience worse. A user or API would have to know when to call which method per token type. And even if you had two methods with two distinct cases, you would have to guard against each method being used against the wrong case with the same lookup of `tokenOwnerAgentId`. Maybe it would be better to have two internal functions which distinguish the two cases and branch in the top-level `sendToken` method.

Reply:
@yrong The thinking behind extending sendToken is that, in both cases, it's dealing with ERC20 tokens that should look the same externally. The only distinction between native vs wrapped tokens is in the Gateway's storage. The intention is "send this ERC20 token to Polkadot" - whether the token is Polkadot-native or not shouldn't matter for the caller.

As Alistair says, a caller would have to figure out which function to call, potentially by querying the `ownerAgentIDs` mapping in the Gateway. We'd then query the same mapping in sendToken anyway, to check if the token has been registered and if the correct function has been called.

Reply:
@alistair-singh Separate internal functions would differ only in the line above, ie. whether we send the token to an agent or burn it. I'm open to separating these cases but  don't think we should until there's a larger difference between them.

Reply:
Yes, that make sense!